fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/LoanApplicationTerms.java

```diff
@@ -228,6 +229,76 @@ public final class LoanApplicationTerms {
     private final boolean enableAccrualActivityPosting;
     private final List<LoanSupportedInterestRefundTypes> supportedInterestRefundTypes;
 
+    public static LoanApplicationTerms assembleFrom(LoanRepaymentScheduleModelData modelData) {
+        return new LoanApplicationTerms(modelData.currency(), // currency
+                modelData.numberOfRepayments(), // loanTermFrequency
+                PeriodFrequencyType.valueOf(modelData.repaymentFrequencyType()), // loanTermPeriodFrequencyType
+                modelData.numberOfRepayments(), // numberOfRepayments
+                modelData.repaymentFrequency(), // repaymentEvery
+                PeriodFrequencyType.valueOf(modelData.repaymentFrequencyType()), // repaymentPeriodFrequencyType
+                null, // nthDay
+                null, // weekDayType
+                null, // amortizationMethod
+                null, // interestMethod
+                modelData.nominalInterestRate(), // interestRatePerPeriod
+                PeriodFrequencyType.valueOf(modelData.repaymentFrequencyType()), // interestRatePeriodFrequencyType
+                modelData.nominalInterestRate(), // annualNominalInterestRate
```

Comment:
We have to handle this part carefully. We have a fix in this file which used wrong interest method.
You can check the history of that change.

https://github.com/apache/fineract/blob/develop/fineract-progressive-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/data/ProgressiveLoanInterestScheduleModel.java#L52

Reply:
@janez89 LoanProductRelatedDetail is collected from the loanApplicationTerms data in the 'generate' method (loanApplicationTerms.toLoanProductRelatedDetail()) and that the AnnualNominalInterestRate should not change if my understanding is correct.